src/lib/index/wb_cache.cpp

```diff
@@ -289,13 +293,7 @@ void IndexWBCache::transact_bufs(uint32_t index_ordinal, IndexBufferPtr const& p
         }
         icp_ctx->add_to_txn_journal(index_ordinal, parent_buf, nullptr, {child_buf}, {});
     } else {
-        icp_ctx->add_to_txn_journal(index_ordinal,          // Ordinal
-                                    child_buf->m_up_buffer, // real up buffer
-                                    new_node_bufs.empty() ? freed_node_bufs[0]->m_up_buffer
-                                                          : new_node_bufs[0]->m_up_buffer, // real in place child
-                                    new_node_bufs,                                         // new node bufs
-                                    freed_node_bufs                                        // free_node_bufs
-        );
+        icp_ctx->add_to_txn_journal(index_ordinal, parent_buf, child_buf, new_node_bufs, freed_node_bufs);
```

Comment:
Just record the original input bufs in journal - the up_buf links will be recovered in `IndexCPContext::process_txn_record`.